gen-ai-rag-vertex-ai-vector-search/vertex.tf

```diff
@@ -0,0 +1,48 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generative AI through Vertex AI
+
+resource "google_vertex_ai_index" "default" {
+  display_name = "index-${local.unique_str}"
+  description  = "Index for storing document embeddings"
+  metadata {
+    config {
+      dimensions                  = 1536
+      approximate_neighbors_count = 0
+      distance_measure_type       = "DOT_PRODUCT_DISTANCE"
+      algorithm_config {
+        brute_force_config {}
+      }
+    }
+  }
+}
+
+resource "google_vertex_ai_index_endpoint" "default" {
+  display_name = "Index endpoint"
+  region       = var.region
+}
+
+resource "google_vertex_ai_index_endpoint_deployed_index" "default" {
+  index_endpoint    = google_vertex_ai_index_endpoint.default.id
+  deployed_index_id = "deployed_endpoint_${local.unique_str}"
+  index             = google_vertex_ai_index.default.id
+  dedicated_resources {
+    machine_spec {
+      machine_type = "n1-standard-16"
```

Comment:
I recommend that we define a variable for this in variables.tf. Users might want to experiment with the machine type 
 to optimize index performance.

Reply:
There are a lot of elements the user might want to change in vertex.tf to optimize performance, not just the deployed index, but all the index configurations. I'll use comments rather than variables to denote this.

Reply:
In https://github.com/GoogleCloudPlatform/architecture-center-samples/pull/5/commits/c84256c177e3633cf4511d5350fe25a39140150c I denoted a tentative comment scheme to differentiate between settings that are Design Considerations, and settings that should be adapted by the user.